account_banking_payment_transfer/model/account_payment.py

```diff
@@ -0,0 +1,259 @@
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, _
+
+
+class PaymentOrder(models.Model):
+    '''
+    Enable extra states for payment exports
+    '''
+    _inherit = 'payment.order'
+
+    date_scheduled = fields.Date(states={
+        'sent': [('readonly', True)],
+        'rejected': [('readonly', True)],
+        'done': [('readonly', True)],
+        })
+    reference = fields.Char(states={
+        'sent': [('readonly', True)],
+        'rejected': [('readonly', True)],
+        'done': [('readonly', True)],
+        })
+    mode = fields.Many2one(states={
+        'sent': [('readonly', True)],
+        'rejected': [('readonly', True)],
+        'done': [('readonly', True)],
+        })
+    state = fields.Selection([
+        ('draft', 'Draft'),
+        ('open', 'Confirmed'),
+        ('cancel', 'Cancelled'),
+        ('sent', 'Sent'),
+        ('rejected', 'Rejected'),
+        ('done', 'Done'),
+        ], string='State')
+    line_ids = fields.One2many(states={
+        'open': [('readonly', True)],
+        'cancel': [('readonly', True)],
+        'sent': [('readonly', True)],
+        'rejected': [('readonly', True)],
+        'done': [('readonly', True)]
+        })
+    user_id = fields.Many2one(states={
+        'sent': [('readonly', True)],
+        'rejected': [('readonly', True)],
+        'done': [('readonly', True)]
+        })
+    date_prefered = fields.Selection(states={
+        'sent': [('readonly', True)],
+        'rejected': [('readonly', True)],
+        'done': [('readonly', True)]
+        })
+    date_sent = fields.Date(string='Send date', readonly=True)
+
+    def action_rejected(self, cr, uid, ids, context=None):
+        return True
+
+    @api.multi
+    def action_done(self):
+        for line in self.line_ids:
+            line.date_done = fields.Date.context_today(self)
+        self.date_done = fields.Date.context_today(self)
+        # state is written in workflow definition
+        return True
+
+    @api.one
+    @api.returns('account.move.line')
+    def _get_transfer_move_lines(self):
+        """
+        Get the transfer move lines (on the transfer account).
+        """
+        for pay_line in self.line_ids:
+            move_line = pay_line.transfer_move_line_id
+            if move_line:
+                return move_line
+        return False
```

Comment:
@alexis-via can you look after this one? It comes from the new api conversion you did, and now that I look at it again, test_done() looks suspicious indeed especially since _get_transfer_move_lines can return False which is not iterable. 

The test suite works, though and it must go through this, I'll check.